[CI] DateTimeUnitTests.testConversion fails non-reproducibly

I cannot reproduce this one.

Failed on 7.0 (bea2628)
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+multijob-unix-compatibility/os=oraclelinux-6/51/console

```
./gradlew :server:unitTest -Dtests.seed=55D6AFA4F0B92ECC \
  -Dtests.class=org.elasticsearch.common.rounding.DateTimeUnitTests \
  -Dtests.method="testConversion" -Dtests.security.manager=true \
  -Dtests.locale=th-TH -Dtests.timezone=SystemV/PST8 \
  -Dcompiler.java=11 -Druntime.java=8
```

```
03:03:05 
03:03:05 Suite: org.elasticsearch.common.rounding.DateTimeUnitTests
03:03:05   1> [2562-03-03T19:03:04,899][INFO ][o.e.c.r.DateTimeUnitTests] [testConversion] before test
03:03:05   1> [2562-03-03T19:03:04,909][INFO ][o.e.c.r.DateTimeUnitTests] [testConversion] after test
03:03:05 FAILURE 0.02s J7 | DateTimeUnitTests.testConversion <<< FAILURES!
03:03:05    > Throwable #1: java.lang.AssertionError: 
03:03:05    > Expected: is <289299900343L>
03:03:05    >      but: was <289303500343L>
03:03:05    > 	at __randomizedtesting.SeedInfo.seed([55D6AFA4F0B92ECC:53DE2A9EDD767158]:0)
03:03:05    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
03:03:05    > 	at org.elasticsearch.common.rounding.DateTimeUnitTests.testConversion(DateTimeUnitTests.java:79)
03:03:05    > 	at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/common/rounding/DateTimeUnitTests.java
+++ b/server/src/test/java/org/elasticsearch/common/rounding/DateTimeUnitTests.java
@@ -19,11 +19,6 @@
 package org.elasticsearch.common.rounding;
 
 import org.elasticsearch.test.ESTestCase;
-import org.joda.time.DateTimeZone;
-
-import java.time.Instant;
-import java.time.ZoneId;
-import java.time.ZonedDateTime;
 
 import static org.elasticsearch.common.rounding.DateTimeUnit.DAY_OF_MONTH;
 import static org.elasticsearch.common.rounding.DateTimeUnit.HOUR_OF_DAY;
@@ -33,7 +28,6 @@ import static org.elasticsearch.common.rounding.DateTimeUnit.QUARTER;
 import static org.elasticsearch.common.rounding.DateTimeUnit.SECOND_OF_MINUTE;
 import static org.elasticsearch.common.rounding.DateTimeUnit.WEEK_OF_WEEKYEAR;
 import static org.elasticsearch.common.rounding.DateTimeUnit.YEAR_OF_CENTURY;
-import static org.hamcrest.Matchers.is;
 
 public class DateTimeUnitTests extends ESTestCase {
 

--- a/server/src/test/java/org/elasticsearch/common/rounding/DateTimeUnitTests.java
+++ b/server/src/test/java/org/elasticsearch/common/rounding/DateTimeUnitTests.java
@@ -65,18 +65,4 @@ public class DateTimeUnitTests extends ESTestCase {
         assertEquals(8, SECOND_OF_MINUTE.id());
         assertEquals(SECOND_OF_MINUTE, DateTimeUnit.resolve((byte) 8));
     }
-
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/39617")
-    public void testConversion() {
-        long millis = randomLongBetween(0, Instant.now().toEpochMilli());
-        DateTimeZone zone = randomDateTimeZone();
-        ZoneId zoneId = zone.toTimeZone().toZoneId();
-
-        int offsetSeconds = zoneId.getRules().getOffset(Instant.ofEpochMilli(millis)).getTotalSeconds();
-        long parsedMillisJavaTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), zoneId)
-            .minusSeconds(offsetSeconds).toInstant().toEpochMilli();
-
-        long parsedMillisJodaTime = zone.convertLocalToUTC(millis, true);
-        assertThat(parsedMillisJavaTime, is(parsedMillisJodaTime));
-    }
 }

